client/starwhale/core/model/model.py

```diff
@@ -830,6 +850,22 @@ def serve(
         svc = cls._get_service(model_config.run.modules, model_src_dir)
         svc.serve(host, port, model_config.name)
 
+    @classmethod
+    def _serve_handler(cls, **kwargs: t.Any) -> None:
+        # https://github.com/star-whale/starwhale/pull/2350
+        # kwargs see cls._gen_model_serving
+        search_modules = kwargs.get("extra_kwargs", {}).get("search_modules", [])
+        if len(search_modules) == 0:
+            raise ValueError("search_modules is empty")
+
+        workdir = kwargs.get("workdir", "")
+        if not workdir:
+            raise ValueError("workdir is empty")
+
+        port = kwargs.get("expose", 8000)
```

Comment:
About the expose port, there are two questions:

1. Should we add EXPOSE_PORT to the script of entrypoint?
2. Is Server should parse the spec and add a rule to expose the port?

Reply:
Like docker's [expose](https://docs.docker.com/engine/reference/builder/#expose)
- The `expose` here does not actually publish the port 
- The `expose` tells the Controller that the job needs to expose a port (the firewall and others should be well configured by the controller, and tell the console the final access URL)
- When getting the expose info, the Controller should proxy the running service (job) if it has no public IP